src/app/shared/issue/duplicateOf/duplicate-of.component.ts

```diff
@@ -24,11 +24,34 @@ export class DuplicateOfComponent implements OnInit {
   @ViewChild(MatSelect) duplicateOfSelection: MatSelect;
   @ViewChild(MatCheckbox) duplicatedCheckbox: MatCheckbox;
 
+  // Max chars visible for a duplicate entry in duplicates dropdown list.
+  readonly MAX_CHAR_VISIBLE_DUPLICATE = 17;
+  // Max chars visible for a non-duplicate entry in duplicates dropdown list.
+  readonly MAX_CHAR_VISIBLE_NONDUPLICATE = 37;
+
   constructor(public issueService: IssueService,
               private errorHandlingService: ErrorHandlingService,
               public permissions: PermissionService) {
   }
 
+  /**
+   * Method checks if the supplied issue requires a tooltip
+   * in the UI as some information may be hidden due to truncation.
+   * @param issue - Displayed issue that may need a tooltip.
+   * @return - false (to enable tooltip) / true (to disable tooltip)
+   */
+  disableTooltip(issue: Issue): boolean {
+    // Duplicated issues contain a postfix that reduce the screen-space
+    // available to display the issue title.
+    if (issue.duplicated) {
+      // If length of issue title is less than threshold, it does not need
+      // a tooltip hence the tooltip can be disabled, vice-versa.
+      return (issue.title.length < this.MAX_CHAR_VISIBLE_DUPLICATE);
+    } else {
+      return (issue.title.length < this.MAX_CHAR_VISIBLE_NONDUPLICATE);
+    }
```

Comment:
It looks like only the constant name is different between the two branches. Try to restructure the code to make that easier to spot. In Java, I would do something like this:
e.g.,
```java
int maxLength = issue.duplicated? this.MAX_CHAR_VISIBLE_DUPLICATE : this.MAX_CHAR_VISIBLE_NONDUPLICATE;
return (issue.title.length < maxLength);
```

Reply:
Edited the function to follow your suggested style 👍